lib/assets/javascripts/views/answers/edit.js

```diff
@@ -0,0 +1,160 @@
+    if (!debounceMap[id]) {
+      debounceMap[id] = debounce(autoSaving);
+    }
+    debounceMap[id](target);
+  };
+  const submitHandler = (e) => {
+    e.preventDefault();
+    const target = $(e.target);
+    const form = target.closest('form');
+    const id = questionId(target);
+    if (debounceMap[id]) {
+      // Cancels the delated execution of autoSaving
+      // (e.g. user clicks the button before the delay is met)
+      debounceMap[id].cancel();
+    }
+    showSavingMessage(target);
+    const formElements = form.serializeArray();
+    const answerId = formElements.find(el => el.name === 'answer[id]');
+
+    if (answerId) {
+      $.ajax({
+        method: form.attr('method'),
+        url: form.attr('action'),
+        data: formElements,
+      }).done((data) => {
+        doneCallback(data, target);
+      }).fail((error) => {
+        failCallback(error, target);
+      });
+    }
+  };
+  const blurHandler = (editor) => {
+    const id = questionId($(`#${editor.id}`));
+    $(`#${editor.id}`).val(editor.getContent()); // Updates target element of editor with its content
+    if (!debounceMap[id]) {
+      debounceMap[id] = debounce(autoSaving);
+    }
+    debounceMap[id]($(`#${editor.id}`));
+  };
+  const focusHandler = (editor) => {
+    const id = questionId($(`#${editor.id}`));
+    if (debounceMap[id]) {
+      /* Cancels the delayed execution of autoSaving, either because user
+       * transitioned from an option_based question to the comment or 
+       * because the target element triggered blur and focus before 
+       * the delayed execution of autoSaving.
+      */
+      debounceMap[id].cancel();
+    }
+  };
+  const formHandlers = ({ jQuery, attachment = 'off' }) => {
+    // Listeners to change and submit for a form
+    jQuery[attachment]('change', changeHandler);
+    jQuery[attachment]('submit', submitHandler);
+  };
+  const editorHandlers = (editor) => {
+    // Listeners to blur and focus events for a tinymce instance
+    editor.on('Blur', () => blurHandler(editor));
+    editor.on('Focus', () => focusHandler(editor));
+  };
+    /*
+    Detaches events from a specific form including its tinymce editor
+    @param { objecg } - jQueryForm to remove events
+  */
+  const detachEventHandlers = (jQueryForm) => {
+    formHandlers({ jQuery: jQueryForm, attachment: 'off' });
+    const tinymceId = jQueryForm.find(`.${editorClass}`).attr('id');
+    Tinymce.destroyEditorById(tinymceId);
+  };
+  /*
+    Attaches events for a specific form including its tinymce editor
+    @param { objecg } - jQueryForm to add events
+  */
+  const attachEventHandlers = (jQueryForm) => {
+    formHandlers({ jQuery: jQueryForm, attachment: 'on' });
+    const tinymceId = jQueryForm.find(`.${editorClass}`).attr('id');
+    Tinymce.init({ selector: `#${tinymceId}` });
+    editorHandlers(Tinymce.findEditorById(tinymceId));
+  };
+  TimeagoFactory.render($('time.timeago'));
+  Tinymce.init({ selector: `.${editorClass}` });
```

Comment:
I see what you mean about tinymce being a headache ;)

Reply:
:) Yep, since the form is refreshed whenever a stale answer is detected, all the events within that form have to re-created again :(